docs/guides/hooks/index.rst

```diff
@@ -0,0 +1,209 @@
+Openedx Hooks Extension Framework
+=================================
+
+To sustain the growth of the Open edX ecosystem, the business rules of the
+platform must be open for extension following the open-closed principle. This
+framework allows developers to do just that without needing to fork and modify
+the main edx-platform repository.
+
+
+Context
+-------
+
+Hooks are predefined places in the edx-platform core where externally defined
+functions can take place. In some cases, those functions can alter what the user
+sees or experiences in the platform. Other cases are informative only. All cases
+are meant to be extended using Open edX plugins and configuration.
+
+Hooks can be of two types, events and filters. Events are in essence signals, in
+that they are sent in specific application places and whose listeners can extend
+functionality. On the other hand Filters are passed data and can act on it
+before this data is put back in the original application flow. In order to allow
+extension developers to use the Events and Filters definitions on their plugins,
+both kinds of hooks are defined in lightweight external libraries.
+
+* `openedx filters`_
+* `openedx events`_
+
+Hooks are designed with stability in mind. The main goal is that developers can
+use them to change the functionality of the platform as needed and still be able
+to migrate to newer open releases with very little to no development effort. In
+the case of the events, this is detailed in the `versioning ADR`_ and the
+`payload ADR`_.
+
+A longer description of the framework and it's history can be found in `OEP 50`_.
+
+.. _OEP 50: https://open-edx-proposals.readthedocs.io/en/latest/oep-0050-hooks-extension-framework.html
+.. _versioning ADR: https://github.com/eduNEXT/openedx-events/blob/main/docs/decisions/0002-events-naming-and-versioning.rst
+.. _payload ADR: https://github.com/eduNEXT/openedx-events/blob/main/docs/decisions/0003-events-payload.rst
+.. _openedx filters: https://github.com/eduNEXT/openedx-filters
+.. _openedx events: https://github.com/eduNEXT/openedx-events
+
+
+How to use
+----------
+
+Using openedx-events in your code is very straight forward. We can consider the
+two possible cases, sending or receiving an event.
+
+
+Receiving events
+^^^^^^^^^^^^^^^^
+
+This is one of the most common use cases for plugins. The edx-platform will send
+and event and you want to react to it in your plugin.
+
+For this you need to:
+
+1. Include openedx-events in your dependencies.
+2. Connect your receiver functions to the signals being sent.
+
+Connecting signals can be done using regular django syntax:
+
+.. code-block:: python
+
+    from openedx_events.learning.signals import SESSION_LOGIN_COMPLETED
+
+    @receiver(COURSE_CERT_CHANGED)
```

Comment:
Should this be `SESSION_LOGIN_COMPLETED` ? It doesn't seem to match the import.

Reply:
Good catch, thank you. It should indeed be `SESSION_LOGIN_COMPLETED`